### ISSUE TITLE
artifactory_ldap_setting -> user_dn_pattern Should be optional

### ISSUE THREAD

**Issue by JustinPullen** [jfrog/terraform-provider-artifactory#362]
**Describe the bug**
user_dn_pattern required when should be optional. 

│ Error: Missing required argument
│ 
│   on main.tf line 61, in resource "artifactory_ldap_setting" "ldap":
│   61: resource "artifactory_ldap_setting" "ldap" {
│ 
│ The argument "user_dn_pattern" is required, but no definition was found.

**Requirements for and issue**

# Configure Artifactory LDAP setting
- [ ] A fully functioning terraform snippet that can be copy&pasted (no outside files or ENV vars unless that's part of the issue)
```hcl
resource "artifactory_ldap_setting" "ldap" {
  key              = "ldap"
  ldap_url         = "ldaps://ldap.com"
}
```

- [ ] Your version of artifactory (you can `curl` it at `$host/artifactory/api/system/version`
>7.0

- [ ] Your version of terraform
1.14

**Expected behavior**
user_dn_pattern should be optional or have a default of ""

**Comment by chb0github** [comment]
This was closed with #355 last week. You don't have repeatable HCL here that shows us what version of the provider you're using

**Comment by chb0github** [comment]
https://github.com/jfrog/terraform-provider-artifactory/blob/master/pkg/artifactory/resource_artifactory_ldap_setting.go#L69-L73

Released with `v2.22.2`
